pkg/api/customization/vsphere/listers.go

```diff
@@ -269,15 +270,18 @@ func listDataStoreClusters(ctx context.Context, finder *find.Finder) ([]string,
 	return data, nil
 }
 
-func listFolders(ctx context.Context, finder *find.Finder) ([]string, error) {
-	folders, err := finder.FolderList(ctx, "vm/*")
+func listFolders(ctx context.Context, finder *find.Finder, dc string) ([]string, error) {
+	folders, err := finder.FolderList(ctx, "*")
 	if err != nil {
 		return nil, err
 	}
 
+	// base case of /<datacenter>/vm is covered by ""
 	data := []string{""}
 	for _, f := range folders {
-		data = append(data, f.InventoryPath)
+		if strings.HasPrefix(f.InventoryPath, fmt.Sprintf("%s/vm/", dc)) {
```

Comment:
nit: don't need to generate string `fmt.Sprintf("%s/vm/", dc)` every time, can move it outside loop

Reply:
great point, moving it.